<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ArterialCalcificationPreProcessor</class>
 <widget class="qMRMLWidget" name="ArterialCalcificationPreProcessor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>362</width>
    <height>440</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="parameterSetLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="parameterSetLabel">
       <property name="text">
        <string>Parameter set:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="parameterSetHorizontalLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <widget class="qMRMLNodeComboBox" name="parameterSetSelector">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>A parameter set groups parameters that define a named study distinctly.</string>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLScriptedModuleNode</string>
          </stringlist>
         </property>
         <property name="showHidden">
          <bool>true</bool>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="baseName">
          <string notr="true">ArterialCalcificationPreProcessor</string>
         </property>
         <property name="noneEnabled">
          <bool>false</bool>
         </property>
         <property name="renameEnabled">
          <bool>true</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="selectNodeUponCreation">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="parameterSetUpdateUIToolButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Update the segment editor and the slice views with the input segmentation and volume if any.</string>
         </property>
         <property name="text">
          <string>â†º</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Inputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="inputSegmentSelectorLabel">
        <property name="text">
         <string>Segmentation:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLSegmentSelectorWidget" name="inputSegmentSelector">
        <property name="toolTip">
         <string>Select an input lumen segment.</string>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="inputVolumeLabel">
        <property name="text">
         <string>Volume:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="inputVolumeSelector">
        <property name="toolTip">
         <string>Select the source volume node of the selected segment.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="hideChildNodeTypes">
         <stringlist notr="true"/>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="editEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="interactionNodeSingletonTag">
         <string notr="true"/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="inputMarginLabel">
        <property name="text">
         <string>Margin:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="inputMarginSpinBox">
        <property name="toolTip">
         <string>Segment calcification around the lumen within this margin.</string>
        </property>
        <property name="suffix">
         <string> mm</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.500000000000000</double>
        </property>
        <property name="value">
         <double>4.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="optionsCollapsibleButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Options</string>
     </property>
     <property name="indicatorAlignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="intensityRangeLabel">
        <property name="text">
         <string>Intensity range:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="intensityRangeHorizontalLayout" stretch="0,0,0,0">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QDoubleSpinBox" name="intensityRangeLowerSpinBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Use this specified value as the lower bound of the calcification intensity range.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="intensityRangeUpperSpinBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Use this specified value as the upper bound of the calcification intensity range.</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="intensityRangeProbeButton">
          <property name="toolTip">
           <string>Probe the intensity range defining the calcifications.

It is calculated as:
 - lower: (medianSegmentHU + maxSegmentHU) / 2.0
 - upper: maxVolumeHU * 0.95

where HU is Houndsfield Units.</string>
          </property>
          <property name="text">
           <string>Probe</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="intensityRangeResetButton">
          <property name="toolTip">
           <string>Reset the intensity range.</string>
          </property>
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="ctkMenuButton" name="applyButton">
     <property name="toolTip">
      <string>Run the algorithm.</string>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkMenuButton</class>
   <extends>QPushButton</extends>
   <header>ctkMenuButton.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSegmentSelectorWidget</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLSegmentSelectorWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>ArterialCalcificationPreProcessor</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputSegmentSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>201</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ArterialCalcificationPreProcessor</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputVolumeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>201</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ArterialCalcificationPreProcessor</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>parameterSetSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
